### PR TITLE
Fix flaky IW test

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: src/kernels/execution/cellExecutionMessageHandler.ts [hunk 1]
```diff
@@ -845,7 +845,7 @@ export class CellExecutionMessageHandler implements IDisposable {
                 name: msg.content.name,
                 text: this.lastUsedStreamOutput.text
             });
-            traceCellMessage(this.cell, `Replace output items ${this.lastUsedStreamOutput.text.substring(0, 100)}`);
+            traceCellMessage(this.cell, `Replace output items '${this.lastUsedStreamOutput.text.substring(0, 100)}'`);
```
Comment (DonJayamanne) [inline src/kernels/execution/cellExecutionMessageHandler.ts:848]: Log the output quoted, so we know where the output starts and ends

File: src/platform/logging/consoleLogger.ts [hunk 2]
```diff
@@ -6,7 +6,9 @@ import { getTimeForLogging } from './util';
 const format = require('format-util') as typeof import('format-util');
 
 function formatMessage(level: string | undefined, message: string, ...data: Arguments): string {
-    return level ? `${level} ${getTimeForLogging()}: ${format(message, ...data)}` : format(message, ...data);
+    const isDataEmpty = [...data].length === 0;
+    const formattedMessage = isDataEmpty ? format(message) : format(message, ...data);
+    return level ? `${level} ${getTimeForLogging()}: ${formattedMessage}` : formattedMessage;
```
Comment (DonJayamanne) [inline src/platform/logging/consoleLogger.ts:11]: All the log outputs end with a `[]`, this change fixes that
see here
```
:00:55 PM.294: Start Test Run current file in interactive window (with cells) []
4:00:55 PM.295: Start Test (completed) Run current file in interactive window (with cells) []
4:00:55 PM.442: Starting interactive window for resource '/tmp/tmp-7950tKpI4dyc2gOP.py' []
4:00:55 PM.442: CreateActiveInterpreterController []
4:00:55 PM.552: Kernel got disposed as a result of session.onDisposed /tmp/tmp-7950ueIE7AykX0cN.py []
4:00:55 PM.553: Dispose KernelExecution []
4:00:55 PM.553: Cell Index:1, of document Interactive-1.interactive with state:1, exec: 1. Execution disposed []
4:00:55 PM.553: Cell Index:1, of document Interactive-1.interactive with state:1, exec: 1. Execution disposed []
4:00:55 PM.553: Cell Index:1, of document Interactive-1.interactive with state:1, exec: 1. Execution disposed []
4:00:55 PM.553: Cell Index:1, of document Interactive-1.interactive with state:1, exec: 1. Execution disposed []
4:00:55 PM.555: Cell Index:1, of document Interactive-1.interactive with state:1, exec: 1. Execution disposed []
4:00:55 PM.555: Dispose KernelExecution []
4:00:55 PM.555: Cell Index:1, of document Interactive-1.interactive with state:1, exec: 1. Execution disposed []
4:00:55 PM.555: Cell Index:1, of document Interactive-1.interactive with state:1, exec: 1. Execution disp
```

File: src/test/datascience/interactiveWindow.vscode.common.test.ts [hunk 3]
```diff
@@ -371,11 +372,27 @@ ${actualCode}
         assert.equal(notebookDocument?.cellCount, 3, `Running a whole file did not split cells`);
 
         // Make sure it output something
-        notebookDocument?.getCells().forEach((c, i) => {
-            if (c.document.uri.scheme === 'vscode-notebook-cell' && c.kind == vscode.NotebookCellKind.Code) {
-                assertHasTextOutputInVSCode(c, `${i}`);
-            }
-        });
+        let lastErrorMessage = 'No output found';
+        await waitForCondition(
```
Comment (DonJayamanne) [inline src/test/datascience/interactiveWindow.vscode.common.test.ts:376]: Found a flaky test, we weren't waiting for the output.
we need to wait for the output.

Comment (amunger) [reply src/test/datascience/interactiveWindow.vscode.common.test.ts:376]: huh, I imagine this would cause other tests to be flaky as well

**Review by amunger — APPROVED** [review]

File: src/test/datascience/interactiveWindow.vscode.common.test.ts (hunk 3 above)
Comment (DonJayamanne) [reply src/test/datascience/interactiveWindow.vscode.common.test.ts:376]: I think most of the other tests do have this, i know for sure that all the nb tests have this as i wrote/updated most of them.
IW is somethign thats been around for a while and I haven't really touched that area much...

**Pull request merged**